src/blob_gc_picker.cc

```diff
@@ -40,7 +42,8 @@ std::unique_ptr<BlobGC> BasicBlobGCPicker::PickBlobGC(
     blob_files.push_back(blob_file.get());
 
     batch_size += blob_file->file_size();
-    if (batch_size >= cf_options_.max_gc_batch_size) break;
+    estimate_output_size += (blob_file->file_size() - blob_file->discardable_size());
+    if (batch_size >= cf_options_.max_gc_batch_size && estimate_output_size >= min_output_threshold) break;
```

Comment:
can we just readahead some files here to decide whether it needs to do next GC? I think calculating `PenddingGCSize` every GC is wasteful.

Reply:
OK, maybe a more convenient way